gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/subject/SubjectManagerV2.java

```diff
@@ -55,7 +58,9 @@ public String generateSubFrom(User user) {
 
     @Override
     public String generateInternalSubFrom(User user) {
-        return user.getSource() + SEPARATOR + user.getExternalId();
+        return Optional.ofNullable(user.getAdditionalInformation())
+                .map(map -> (String) map.get(GIO_INTERNAL_SUB))
+                .orElseGet(() -> user.getSource() + SEPARATOR + user.getExternalId());
```

Comment:
**discussion**: I don't think this is the place to test and fallback on the gio_internal_sub :thinking: And I'm not even sure that we have to fallback to the additional property as it is very specific to the ExtensionGrant plugin... I'm afraid this change will bring more issues in the future...

Reply:
Impleneted accordingly